package.json

```diff
@@ -53,6 +53,11 @@
     "license-check": "jsgl --local ."
   },
   "dependencies": {
+    "@google-cloud/common": "^0.16.0",
+    "@types/express": "^4.11.0",
```

Comment:
huh, it seems odd that `@types/` packages are dependencies.

Reply:
Oh, I thought this was the opposite: from deps to devDeps. Yeah, this doesn't seem right. These modules shouldn't be installed in users' production binaries.

Reply:
I thought it was strange too, but the types for express etc. are needed because express is referenced in `index.d.ts`.

Issue [Microsoft/types-publisher #81](https://github.com/Microsoft/types-publisher/issues/81) gives a good overview of this issue.  Essentially, library authors should have type dependencies in `dependencies` while app authors should have it in `devDependencies`.

Reply:
That issue is specific to `@types` packages. If we have to do this to all TypeScript projects, all the apps will end up with having all sorts of `@types` libraries in their deps, which seems weird.

What exactly is the error when they are in devDeps?

Reply:
Will it be possible to re-export the required types from index.ts?

Like

```ts
export {Request, Response} from 'express';
```

Reply:
I tried it, and it doesn't work.

Reply:
Do we need those types because the `ErrorReporting` class has public methods such as `express()`, `hapi()`, etc? https://github.com/googleapis/nodejs-error-reporting/blob/master/src/index.ts#L108

Maybe this is a limitation in TS but I don't really like the fact that users need to install a lot of `@types` packages that they don't even need. For example, users who uses pure JS, users who only use express, not hapi, etc.

What do you think, @ofrobots ?

Reply:
@jinwoo It's necessary when the function signatures are written that way. I think it's acceptable to put @types packages in dependencies only if we already depend on those packages, but in this case we explicitly do not.

My thought is that you (@DominicKramer) might just need to rewrite the necessary types locally to describe these function types.

Reply:
Yes.  Essentially the library uses the types exposed by @types/hapi, @types/restify, etc. when defining plugins for those libraries.  However, hapi, restify, etc. are not needed directly.

Also, I think having the user require an `@types` module would be preferred over manually specifying the types used in the `@types` module locally.

Reply:
Note the `@types` dependencies are on the order of 300Kb in size.